Standardizing parameters in sleep models

bruAristimunha opened the pull request:
Hello everyone,

I hope this message finds you well. I wanted to share some observations I made today while studying sleep models. For your reference, I have shared a [Google Spreadsheets](https://docs.google.com/spreadsheets/d/1WZnm5DfgVM3CeRke1SEGHVSkMB1Tt-wkt34fpkfVkDc/edit?usp=sharing) link that includes the table that I made.

I would like to propose some changes that could potentially help disambiguate the parameters of these models, such as renaming "`in_chans`" to "`n_channels`" and "`n_conv_channels`" and "`n_conv_chans`" to simply "`n_conv_channels`".

To this end, I have created a function `check_deprecation_warning` that can help disambiguate other models, particularly in the BCI models, where there is more room for improvement.
 
Regarding the yellow variable in the table, I was initially unsure about its disambiguation. However, upon further analysis of the code's logic, I decided not to change "max_pooling" in the second instance from "max pooling in vector size" (?).

It seems to me that there is a need for improved documentation on how we describe each parameter and better internal standardisation on some internal variables. However, I understand this is not the appropriate PR for this matter. 

I would be grateful if you could review my proposed changes and provide any feedback you may have. Additionally, if you identify any other variables that can benefit from disambiguation, please feel free to share your comments.

Thank you very much in advance for your attention and assistance!

robintibor commented on braindecode/models/sleep_stager_blanco_2020.py:
```diff
@@ -1,11 +1,16 @@
 
+from .util import check_deprecation_warning
+
+pair_new_names_blanco = {"n_conv_chans": "n_conv_channels"}
```
I woould move this to inside the constructor, even to inside the if... no need to have this so global I think

robintibor commented on braindecode/models/sleep_stager_blanco_2020.py:
```diff
@@ -48,44 +56,55 @@ class SleepStagerBlanco2020(nn.Module):
         self.n_channels = n_channels
+        self.n_conv_channels = n_conv_channels
+
+        if kwargs:
```
```suggestion
        if kwargs:
            pair_new_names_blanco = {"n_conv_chans": "n_conv_channels"}
```

robintibor commented on braindecode/models/util.py:
```diff
@@ -1,13 +1,55 @@
+    contains the deprecated parameters.
+
+    """
+    if not all([k in old_new_name_pairs for k in kwargs]) and \
```
I feel better make this in a loop and then give a specific error, like which one is not valid... or you could also use set intersection etc instead of loop, but definitely feel better if error warning says which one is not correct

robintibor commented on test/unit_tests/models/test_util.py:
```diff
@@ -11,6 +11,17 @@
+def test_check_deprecation_warning():
+    in_chans = 2
+
+    with pytest.warns(
```
nice test

bruAristimunha replied:
Thank you @robintibor =)

bruAristimunha commented:
Hi @robintibor!

I updated the code to use set and moved the variable into the models as you suggested! Thank you so much for the quick and good review :)

I think @agramfort, already make his pre-review with the comment: https://github.com/braindecode/braindecode/issues/286#issuecomment-1461537272

But, I didn't fully understand how to change from kwargs to ",*".

bruAristimunha commented:
Hi @agramfort

I've taken the time to study [PEP 3102](https://peps.python.org/pep-3102/), and now I have a better understanding of the use of the "," parameter to indicate the end of positional parameters.

However, I'm still a bit unsure if this is a good approach for our particular case. From what I can see, it would only be useful if we were hardcoding all the old and new parameters into the functions. Am I on the right track? 

I think what I'm proposing here is a little simpler and more readable. What do you think?

Thank you very much for your time and consideration!

bruAristimunha commented:
If I understand correctly, the idea is to change to:

```python 
def __init__(self, in_channels, n_channels, sfreq, n_conv_channels=20, input_size_s=30, n_classes=5, 
n_groups=2, max_pool_size=2, dropout=0.5, apply_batch_norm=False,  return_feats=False, *, in_chans=None):
```

Instead of the code I proposed in the PR with kwargs.

agramfort reviewed (commented):
looks reasonable. I am surprised you don't have to any other test. To me an uncaught warning should raise an error in a test and maybe even when building the doc. I am assume their are many other places where you need to do the changes no?
